Domain/Services/Communication/BaseResponse.cs

```diff
@@ -1,14 +1,22 @@
 namespace WebAPI.API.Domain.Services.Communication
 {
-    public abstract class BaseResponse
+    public abstract class BaseResponse<T>
     {
-        public bool Success { get; protected set; }
-        public string Message { get; protected set; }
+        public bool Success { get; private set; }
+        public string Message { get; private set; }
+        public T Resource { get; private set; }
 
-        public BaseResponse(bool success, string message)
+        protected BaseResponse(T resource)
         {
-            Success = success;
+            Success = true;
+            Message = string.Empty;
+            Resource = resource;
+        }
+        protected BaseResponse(string message)
+        {
+            Success = false;
             Message = message;
+            Resource = default;
```

Comment:
This is a generalization of the baseResponse to allow usto return any object of type T in the base class instead of having each derived class specifying the type explicitly.  It makes the base class more universally useful.